Pet of a given type

Jessica-Mach opened the pull request:
User Story: Pets of a Given Type

- set up a schema for adoptable pets
- updated the Seeder to include adoptable pets
- added adoptable pets to the data.json
- set up an api endpoint `/api/v1/petTypes/pets/:type`
- when a user clicks on a type of pet, the user gets all pets of that type
- the list of pets shows name, picture, age, and vaccination status
- styling will be treated as a separate feature

jackeymason committed:
Schema and seeder updated. Some 4 to 2 space edits.

Jessica-Mach committed:
Added AdoptablePet model and api/v1/petTypes/pets/:id endpoint

jackeymason committed:
Adoptable pets story

Jessica-Mach committed:
No changes made, but git needs me to commit this one file because I looked at it.

Jessica-Mach committed:
Merged an accidental update to PetTypeTile with Jack's changes.

bkincart commented on client/components/AdoptablePetsOfAType.js:
```diff
@@ -0,0 +1,52 @@
+      {pets}
+    </div>
+  )
+
```
please remove any extra whitespace lines here!

bkincart commented on client/components/AdoptablePetsOfAType.js:
```diff
@@ -0,0 +1,52 @@
+
+  const pets = petType.adoptablePets.map( pet => {
+    return <PetTile
+      key = {pet.id}
```
indent lines 30-32

bkincart commented on client/components/PetTile.js:
```diff
@@ -0,0 +1,19 @@
+        <img src={imageUrl} alt={props.type} />
+      </Link>
+      <p>Age: {age}</p>
+      <p>Vaccination Status: {vaccinationStatus? 'YES' : 'NO'}</p>
```
Nice ternary :) for clarity, let's add a space between `vaccinationStatus` and the question mark!

bkincart commented on server/Models/PetType.js:
```diff
@@ -37,6 +37,23 @@ class PetType {
         }
     }
+
+    async adoptablePets(){
```
BEAUTIFUL association here :)

bkincart commented on server/db/Seeder.js:
```diff
@@ -10,18 +10,28 @@ class Seeder {
     let contents = JSON.parse(loadedData);
-    
+
     //console.log(contents.data.petTypes[0]);
```
Let's remove this extra console.log at this point :)

bkincart reviewed (changes requested):
Really beautiful work here! A few small changes then okay to merge.

Jessica-Mach committed:
Adjusted spacing and syntax and removed console.logs

Jessica-Mach commented:
Made the Following Changes Based on Feedback:

- removed extra spaces from AdoptablePetsOfAType.js
- indented lines 30-32 in AdoptablePetsOfAType.js
- added a space to the ternary in PetTile.js
- removed console.logs from the Seeder
